autosubmit/git/autosubmit_git.py

```diff
@@ -260,6 +261,19 @@ def clone_repository(as_conf, force, hpcarch):
 
         return True
 
+    @staticmethod
+    def is_git_repo(git_repo: str) -> bool:
+        git_repo = git_repo.lower().strip()
+
+        git_url_pattern = re.compile(
+                r'^(https?://[a-zA-Z0-9.-]+/[a-zA-Z0-9.-]+/[a-zA-Z0-9.-]+\.git|git@[a-zA-Z0-9.-]+:[a-zA-Z0-9./-]+\.git)$'
+        )
+        file_url_pattern = re.compile(
+            r'^file://.+$'
+        )
```

Comment:
Maybe we can move this regular expression so that it's compiled just once, instead of for every method call. And I think we can use a single RE.